client/machine.go

```diff
@@ -27,13 +28,25 @@ func (m *Machine) Get(systemID string) (ma *entity.Machine, err error) {
 }
 
 // Update machine.
-func (m *Machine) Update(systemID string, machineParams *entity.MachineParams, powerParams map[string]string) (ma *entity.Machine, err error) {
+func (m *Machine) Update(systemID string, machineParams *entity.MachineParams, powerParams map[string]interface{}) (ma *entity.Machine, err error) {
 	qsp, err := query.Values(machineParams)
 	if err != nil {
 		return
 	}
 	for k, v := range powerParams {
-		qsp.Add(k, v)
+		val := reflect.ValueOf(v)
+		if val.Kind() == reflect.String {
+			qsp.Add(k, v.(string))
+		} else if val.Kind() == reflect.Slice &&
+			val.Len() > 0 &&
+			val.Index(0).Kind() == reflect.String {
```

Comment:
I am not aware of all possible variations of power parameters and how Terraform passes parameters from the configuration, but I don't think this code will work for values of type `boolean`, `int` or `[]int` and others.

So imagine a test like this:
```go
func TestConverter(t *testing.T) {
	testcases := map[string]struct {
		in  map[string]interface{}
		out string
	}{
		"strings": {
			in: map[string]interface{}{
				"a": "b",
				"c": "d",
			},
			out: "a=b&c=d",
		},
		"strings array": {
			in: map[string]interface{}{
				"a": []string{"b", "c"},
			},
			out: "a=b&a=c",
		},
		"numbers": {
			in: map[string]interface{}{
				"a": 1,
				"b": 2,
			},
			out: "a=1&b=2",
		},
		"numbers array": {
			in: map[string]interface{}{
				"a": []int{1, 2},
			},
			out: "a=1&a=2",
		},
	}

	for name, tc := range testcases {
		tc := tc

		t.Run(name, func(t *testing.T) {
			t.Parallel()
			res := convert(tc.in).Encode()
			if res != tc.out {
				t.Fatalf("want:%s, got:%s", tc.out, res)
			}
		})
	}
}
```

Reply:
If other types are possible, then maybe something like this will be better in terms of extendability?

```go
func convert(powerParams map[string]interface{}) url.Values {
	qsp := url.Values{}

	for k, v := range powerParams {
		val := reflect.ValueOf(v)
		switch val.Kind() {
		case reflect.String:
			qsp.Add(k, v.(string))
		case reflect.Int:
			qsp.Add(k, fmt.Sprintf("%d", v.(int)))
		case reflect.Slice:
			if val.Len() == 0 {
				continue
			}
			items := make([]string, val.Len())
			for i := 0; i < val.Len(); i++ {
				switch val.Index(0).Kind() {
				case reflect.String:
					items[i] = val.Index(i).String()
				case reflect.Int:
					items[i] = fmt.Sprintf("%d", val.Index(i).Int())
				}
			}
			qsp[k] = items
		}
	}
	return qsp
}
```

Reply:
The newly created util function covers the following types: `string`, `int`, `bool`, `[]string`, `[]int`, `[]bool`